Bluetooth: att: Retry ATT request if security elevation is in progress

MariuszSkamra opened the pull request:
If the conn security elevation is already in progress, retry the ATT
request if failed due to security reasons.

This adds tests for automatic security elevation and retry on security
errors. This includes two tests:
1. where the security is elevated by the central
2. the security is elevated by security request, so pairing is already
   in progress.

Depends on: https://github.com/zephyrproject-rtos/zephyr/pull/64563 (once merged, this will link the testlib library)

MariuszSkamra committed:
tests: Bluetooth: Add tests for ATT automatic security elevation

This adds tests for automatic security elevation and retry on security
errors. This includes two tests:
1. where the security is elevated by the central
2. the security is elevated by security request, so pairing is already
   in progress.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

MariuszSkamra committed:
Bluetooth: att: Retry ATT request if security elevation is in progress

If the conn security elevation is already in progress, retry the ATT
request if failed due to security reasons.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

sjanc commented on subsys/bluetooth/host/smp.c:
```diff
@@ -5802,12 +5802,12 @@ int bt_smp_start_security(struct bt_conn *conn)
 		/* pairing is in progress */
 		if (atomic_test_bit(smp->flags, SMP_FLAG_PAIRING)) {
-			return -EBUSY;
+			return -EINPROGRESS;
```
not sure if we should change this just for sake of 'better' return code  (ie it may break existing application that is already checking against EBUSY)

MariuszSkamra committed:
tests: bsim: testlib: Fix compilation error

Fixes compilation error seen if CONFIG_BT_EATT is unset

testlib/src/att_read.c:179:73: error: expected expression before ‘,’ token
179 |               IF_ENABLED(CONFIG_BT_EATT, (.chan_opt = bearer)),
    |                                                               ^

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>
